generator/internal/client_generator.cc

```diff
@@ -340,6 +340,9 @@ Status ClientGenerator::GenerateCc() {
 
   // includes
   CcPrint("\n");
+  // We will access a deprecated field when a google::api::method_signature
```

Comment:
This suggests that you know when to add this `#include` and when to omit it.  Your call.

Reply:
Done.  Which means this now only affects the `GoldenKitchenSinkClient`, where we have manufactured the circumstance.  No real services currently have emitted overloads using deprecated fields.  PTAL.